data/registry_test.go

```diff
@@ -13,6 +13,11 @@ var dummyCollectorConstructor = func(i interface{}, params CollectorParams) (Col
 }
 
 func TestRegister(t *testing.T) {
+	defer func() {
```

Comment:
We use a global registry, so this needs to be cleaned up after each test. This does that.